.github/workflows/prepare-rebase-on-upstream.yml

```
@@ -0,0 +1,52 @@
+name: Prepare rebase on upstream
+on:
+  workflow_dispatch:
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    permissions: write-all
```

Nit: Now that we're using a deploy key for permissions, should we narrow down the permissions given here? It sounds like we're only using the token for the PR related operations:

```suggestion
    permissions:
      pull-requests: write
```